tests/copp/test_copp.py

```diff
@@ -0,0 +1,199 @@
+"""
+Tests the COPP feature in SONiC.
+
+Notes:
+    These test cases require that a special RPC syncd image is installed on the
+    DUT. You can either pre-install this image and run the test normally, or
+    specify the `--swap-syncd` flag from the command line to have the test fetch
+    the RPC image and install it before the test runs.
+
+    These test cases limit the PPS of all trap groups to 600. This is done to ensure
+    that the PTF can send traffic fast enough to trigger the policer. In order to validate
+    higher rate limits, a physical traffic generator is needed, which is beyond the scope
+    of these test cases.
+
+Parameters:
+    --nn_target_port <port> (int): Which port you want the test to send traffic
+        to. Default is 3.
+
+        Note that this is not the same as the interface name. For example, Ethernet12
+        may not be the 12th port in your system depending on the HWSKU under test.
+
+    --pkt_tx_count <n> (int): How many packets to send during each individual test case.
+        Default is 100000.
+
+    --swap_syncd: Used to install the RPC syncd image before running the tests. Default
+        is disabled.
+"""
+
+import time
+from collections import namedtuple
+
+import pytest
+from copp import copp_utils
+from ptf_runner import ptf_runner
+from common.system_utils import docker
+from common.broadcom_data import is_broadcom_device
+
+_COPPTestParameters = namedtuple("_COPPTestParameters",
+                                 ["nn_target_port",
+                                  "pkt_tx_count",
+                                  "swap_syncd",
+                                  "topo"])
+_SUPPORTED_TOPOS = ["ptf32", "ptf64", "t1", "t1-lag"]
+_TEST_RATE_LIMIT = 600
+
+class TestCOPP(object):
+    """
+    Tests basic COPP functionality in SONiC.
+    """
+
+    @pytest.mark.parametrize("protocol", ["ARP",
+                                          "IP2ME",
+                                          "SNMP",
+                                          "SSH"])
+    def test_policer(self, protocol, duthost, ptfhost, _copp_testbed):
+        """
+        Validates that rate-limited COPP groups work as expected.
+
+        Checks that the policer enforces the rate limit for protocols
+        that have a set rate limit.
+        """
+
+        if protocol == "ARP" and is_broadcom_device(duthost):
+            pytest.xfail("ARP policy disabled on BRCM devices due to SAI bug")
```

Comment:
This is not universally true. Right? 201811 doesn't subject to this problem and this test is able to run against 201811 images. Should we also check image version here?

Reply:
That's a good point, I'll add a version check!

Reply:
Just added, let me know what you think!